Golem/Yagna/Provider.cs

```diff
@@ -109,10 +109,10 @@ public class Provider : IProvider
         private Process? ProviderProcess { get; set; }
         private SemaphoreSlim ProcLock { get; } = new SemaphoreSlim(1, 1);
 
-        public Provider(string golemPath, string? dataDir, ILoggerFactory? loggerFactory = null)
+        public Provider(string golemPath, string? dataDir, ILoggerFactory loggerFactory)
```

Comment:
Another logger problem

Reply:
LoggerFactory should follow the null object pattern instead of mixing nullable and null object approaches